### ISSUE TITLE
Support for creating WASI projects

### ISSUE THREAD

**Issue by zhuowei** [wasdk/WebAssemblyStudio#415]
Will there be support for building WebAssembly projects that use the [WASI sdk](https://wasi.dev)?

WASI is a new effort to create a standard set of syscalls for WebAssembly code, so the same wasm file can run in different WebAssembly environments.

A WASI intro project seems like a good follow up project to the existing "hello world in C" project. That project has the user create their own syscall bindings, so introducing WASI afterwards would be a logical progression, as the user now understands what WASI does.

I'm guessing this would probably require two components:

- add a new compiler service that uses the WASI sdk as the sysroot
- add some way to run the resulting binary in the browser, probably based on the WASI polyfill.

Would this be a good addition to WebAssembly Studio? If so, how can I help?

**Comment by yurydelendik** [comment]
> Will there be support for building WebAssembly projects that use the WASI sdk?

Yes, there are plans to move clang compiler service to wasi-sdk.

>  If so, how can I help?

The https://github.com/wasdk/WebAssemblyStudio/wiki/Micro-services#clang-compiler-service needs to be updated and examples added. Yes, and help will be welcome.
